sentry/src/main/java/io/sentry/instrumentation/file/FileIOSpanManager.java

```diff
@@ -0,0 +1,110 @@
+package io.sentry.instrumentation.file;
+
+import io.sentry.IHub;
+import io.sentry.ISpan;
+import io.sentry.SpanStatus;
+import io.sentry.util.Pair;
+import io.sentry.util.Platform;
+import io.sentry.util.StringUtils;
+import java.io.Closeable;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+final class FileIOSpanManager {
+
+  private final @Nullable ISpan currentSpan;
+  private final @Nullable File file;
+  private final @NotNull IHub hub;
+
+  private @Nullable Throwable throwable = null;
+  private @NotNull SpanStatus spanStatus = SpanStatus.OK;
+  private long byteCount;
+
+  static @Nullable ISpan startSpan(final @NotNull IHub hub, final @NotNull String op) {
```

Comment:
I'm a bit lost here, we have a static method within an instance class, but also a ctor that takes a `hub`, mind elaborating the use case?

Reply:
yeah, if you take a look at how it's used, we need this static method, because we start the span *before* the FIOS class is instantiated to also measure how much time the ctor of FIOS will take. the hub is propagated here because we need to test this.

As for the ctor, the only reason to have it there is because we are using it to retrieve the options to check if `sendDefaultPii` is enabled in `finishSpan` (which we also want to test)

Reply:
gotcha, the method is more a DRY instead of having the very same method twice in both classes.

wondering if it makes sense to pass the `sendDefaultPii` via ctor, instead of the whole `hub`

Reply:
Yeah good point, we could do that certainly